editor/src/components/canvas/canvas-strategies/strategies/reparent-metastrategy.tsx

```diff
@@ -87,13 +87,25 @@ export function getApplicableReparentFactories(
       case 'REPARENT_AS_STATIC': {
         const fitness = result.isFallback ? 2 : 3
 
-        const targetParentDisplayType =
-          MetadataUtils.findElementByElementPath(
+        const childrenPaths = replaceContentAffectingPathsWithTheirChildrenRecursive(
+          canvasState.startingMetadata,
+          canvasState.startingAllElementProps,
+          MetadataUtils.getChildrenPathsUnordered(
             canvasState.startingMetadata,
             result.target.newParent,
-          )?.specialSizeMeasurements.display === 'flex'
-            ? 'flex'
-            : 'flow'
+          ),
+        )
+
+        const parentLayouSystems = mapDropNulls(
+          (path) => MetadataUtils.findElementByElementPath(canvasState.startingMetadata, path),
+          childrenPaths,
+        ).map((instance) => instance.specialSizeMeasurements.parentLayoutSystem)
+
+        if (!allElemsEqual(parentLayouSystems)) {
+          throw new Error('All children should have the same parent layout system')
```

Comment:
I think this is not an invariant, if the target element in question is a component that wraps children in a weird way we can end up with children that have different actual DOM parents

Reply:
actually, why can't this use findLayoutSystemForChildren ?

Reply:
we can use it, implemented in https://github.com/concrete-utopia/utopia/pull/3334/commits/9061d33830fd6e528efad4f611ca641d1a7ec414